packages/tradershub/src/features/cfd/modals/MT5PasswordModal/MT5CreateAccountButton.tsx

```diff
@@ -1,26 +1,38 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useQueryParams } from '@/hooks';
 import { Button } from '@deriv-com/ui';
 import useMT5AccountHandler from '../../../../hooks/useMT5AccountHandler';
-import { validPassword } from '../../../../utils/password';
 
 type TCreateAccountButtonProps = {
     buttonText: string;
     password: string;
 };
 
 const MT5CreateAccountButton = ({ buttonText, password }: TCreateAccountButtonProps) => {
-    const { createMT5AccountLoading, handleSubmit, tradingPlatformPasswordChangeLoading } = useMT5AccountHandler();
+    const {
+        createMT5AccountLoading,
+        doesNotMeetPasswordPolicy,
+        handleSubmit,
+        tradingPlatformPasswordChangeLoading,
+        createMT5AccountStatus,
+    } = useMT5AccountHandler();
+    const { openModal } = useQueryParams();
+
     const isLoading = tradingPlatformPasswordChangeLoading || createMT5AccountLoading;
-    const isDisabled = !password || isLoading || !validPassword(password);
+    const isDisabled = !password || isLoading;
+
+    useEffect(() => {
+        if (doesNotMeetPasswordPolicy) {
+            openModal('MT5ChangePasswordModal');
```

Comment:
```suggestion
            openModal('MT5ChangePasswordModal');
            return;
```
either add an early return here or use else if. If someone were to change these conditions in the future where they have the chance to happen at both time the next statement would probably override the previous one